### PR TITLE
SLING-12218 - Limit queue size

### PR DESCRIPTION
**Pull request by cschneider**

### CODE REVIEW

File: src/main/java/org/apache/sling/distribution/journal/queue/impl/PubQueueProviderImpl.java [hunk 1]
```diff
@@ -201,6 +202,26 @@ public OffsetQueue<DistributionQueueItem> getOffsetQueue(String pubAgentName, lo
...
     }
+    
+
+    public int getMaxQueueSize(String pubAgentName) {
+        Optional<Long> minOffset = getMinOffset(pubAgentName);
+        if (minOffset.isPresent()) {
+            return getOffsetQueue(pubAgentName, minOffset.get()).getMinOffsetQueue(minOffset.get()).getSize();
+        } else {
+            return 0;
+        }
```
Comment (danilobanjac) [inline src/main/java/org/apache/sling/distribution/journal/queue/impl/PubQueueProviderImpl.java:213]:
```suggestion
       return minOffset.map(offset -> getOffsetQueue(pubAgentName, offset).getMinOffsetQueue(offset).getSize())
                .orElse(0);
```

Comment (danilobanjac) [reply src/main/java/org/apache/sling/distribution/journal/queue/impl/PubQueueProviderImpl.java:213]: This is only if you want, optional has helper methods to perform basic conditional check.

**Review by danilobanjac — APPROVED** [review]
LGTM

**Comment by cschneider** [comment]
@danilobanjac I have added the logic to delay processing when we are near the queue size limit. 
For now I implemented a very simple delay though.

**Pull request merged**